### ISSUE TITLE
Zenmap showing results at end of scan only, even when using "--max-hostgroup N"

### ISSUE THREAD

**Issue by jjgh** [nmap/nmap#469]
Using Zenmap I noticed that scan results are shown only when the scan is completed, no matter I use the "--max-hostgroup 64" parameter. 

Isn't this parameter supposed to force Nmap to process only 64 hosts in parallel, and show the results as soon as the scanning of those 64 is completed?

**Comment by dmiller-nmap** [comment]
The trouble is that Zenmap doesn't process Nmap output until after the scan completes. It shouldn't be out of the question to instead process output as a stream, loading each hostgroup as it is finished.

**Comment by jjgh** [comment]
I suspected that.. :/
IMHO this should really be a high priority change,  because it may affect new users opinions on Nmap and possibly their support. 
Thank you for your answer.

**Comment by fyodor** [comment]
Yeah, Zenmap shows the normal Nmap output streaming through as it goes, but the parts which rely on parsing the XML (such as the host list) don't show until the whole scan completes.  It would probably be better to stream the XML as we go, but it would be a fairly big architectural change.  So we're keeping this issue open for now in case we or someone finds time to do it.

**Comment by illtellyoulater** [comment]
> It would probably be better to stream the XML as we go

Yep, that would work great!
Perhaps this could be added to a roadmap or any other place where it's more exposed to devs willing to help?
